### PR TITLE
Documentation Lesson.

### PR DESCRIPTION
**Pull request by pjhampton**
References: [#7](https://github.com/doomspork/elixir-school/issues/7)

**Commit details:**

* Assets folder ~ containing screenshots of ExDoc example.
* Documentation lesson (documentation.md)

**Warnings:**

* Images for the documentation lesson won't show up in localhost because it uses the `{{ site.url }}` interpolated variable. Advice followed from [Jekyll Docs](http://jekyllrb.com/docs/posts/#including-images-and-resources).

### CODE REVIEW

File: lessons/basics/documentation.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+- [Best Practice](#best-practice)
+
+
+## Annotation
+
+How much we comment and what makes quality documentation remains a contentious issue within the programming world. However, we can all agree that documentation is important for ourselves and those working with our codebase. 
+
+Elixir treats documentation as a *first-class citizen*, offering various functions to access and generate documentation for your projects. The Elixir core provides us with many different attributes to annotate a codebase. Let's look at 3 ways:
+
+  - `#`~ For inline documentation.
```
Comment (doomspork) [inline lessons/basics/documentation.md]: In place of `~` could you use `-`?  That'll be a bit more consistent with other lessons.

**Comment by pjhampton** [comment]
@doomspork I assume you are just talking about lines 29-31? The console output I just left the way it was. Happy to change that to though.

File: lessons/basics/documentation.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+```
+
+We (or others) can access this module documentation using the `h` helper function within IEx. 
+
+```bash
```
Comment (doomspork) [inline lessons/basics/documentation.md]: For IEx blocks I use elixir syntax highlighting, disable the prompt line number, and skip the bash portion: 

```elixir
iex> c("greeter.ex")
[Greeter]
iex> h Greeter

                                    Greeter

Provides a function hello/1 to greet a human
```

Configuring the prompt is pretty straight forward:

```elixir
iex(2)> IEx.configure(default_prompt: "%prefix>")
:ok
iex> case :ok do
...> :ok -> "Hi"
...> _ -> "Other"
...> end
"Hi"
```

File: lessons/basics/documentation.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+We specify the `only: :dev` key-value pair as we don't want to download and compile these dependencies in a production environment. But why Earmark? Earmark is an markdown parser for the elixir programming language that ExDoc utilises to turn our documentation within `@moduledoc` and `@doc` to beautiful looking HTML. 
+
+It is worth noting at this point, that you are not forced to use Earmark. You can change the markup tool to others such as Pandoc or Hoedown; however you will need to do a little more configuration which you can read about [here](https://github.com/elixir-lang/ex_doc#changing-the-markdown-tool). For this tutorial, we'll just stick with Earmark.
+
+### Generating Documentation
+
+Carrying on, from the command line run the following two commands:
+
+```bash
+greet_everyone/$ ~ mix deps.get # gets ExDoc + Earmark.
```
Comment (doomspork) [inline lessons/basics/documentation.md]: I've been using `$` to denote when in the shell versus `iex>` for IEx; I've _tried_ to be consistent throughout the lessons.  

Any objections to changing these lines to something like, to match the other lessons?

```shell
$ mix deps.get
$ mix docs
```

**Comment by doomspork** [comment]
Howdy @pjhampton, thanks again for putting this together!

I left some comments about the IEx and Shell blocks.  If you're okay with the suggested changes can you apply them throughout the document?  I didn't to comment on each instance.

I'll be home shortly and then I can give this one final look over before testing it out and deploying.

File: lessons/basics/documentation.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+- [Best Practice](#best-practice)
+
+
+## Annotation
+
+How much we comment and what makes quality documentation remains a contentious issue within the programming world. However, we can all agree that documentation is important for ourselves and those working with our codebase. 
+
+Elixir treats documentation as a *first-class citizen*, offering various functions to access and generate documentation for your projects. The Elixir core provides us with many different attributes to annotate a codebase. Let's look at 3 ways:
+
+  - `#`- For inline documentation.
```
Comment (doomspork) [inline lessons/basics/documentation.md]: There should be a space before this first `-`

File: lessons/basics/documentation.md (hunk 3 above)
Comment (pjhampton) [inline lessons/basics/documentation.md]: No objections!

File: lessons/basics/documentation.md (hunk 2 above)
Comment (pjhampton) [inline lessons/basics/documentation.md]: Gotcha!

**Comment by doomspork** [comment]
@pjhampton if you want to display the images locally you need to update `_config.yml` by changing `url` to localhost:

```yml
# Setup
title: Elixir School
description: Lessons in the Fundamentals of Elixir
baseurl: /
url: http://127.0.0.1:4000
permalink: pretty
```

**Comment by doomspork** [comment]
For those involved with translations this PR includes a new Basics lesson: Documentation.

cc @milmazz, @erickgnavar, @cizixs, @3100

**Comment by doomspork** [comment]
This looks great.  Thanks again @pjhampton :+1:

**Pull request merged**